libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/MetadataSyncDownTarget.java

```diff
@@ -86,6 +87,10 @@ public JSONObject asJSON() throws JSONException {
     public JSONArray startFetch(SyncManager syncManager, long maxTimeStamp) throws IOException, JSONException {
         final RestRequest request = RestRequest.getRequestForDescribe(syncManager.apiVersion, objectType);
         final RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+        final JSONObject responseJSON = response.asJSONObject();
+        if (responseJSON != null) {
+            responseJSON.put(Constants.ID, objectType);
```

Comment:
The soup we store that data in has an index on Id already?

Reply:
I thought we did some work to auto-add `Id` and `LastModifiedDate` to the index specs if it was missing? I'll check.

Reply:
I have created the high-level sync helper for metadata yet. The query on layout is based on `objectType` and `layoutType` already.

Reply:
The default `externalId` is `Id`, so this should work.